### DISCUSSION TITLE
Separate shared and user storage?

### DISCUSSION THREAD

**Discussion by alimanfoo** [nebari-dev/nebari#706] [Q&A]
Currently IIRC both user home directories and the shared group directories are backed by the same NFS volume.

Is there any way to provision storage for shared folders and for user home directories separately? 

I'm anticipating that some users may not be careful and will store too much data in their home directories, which could currently then affect other users by causing everyone to run out of space.

Ideally I would like to be able to allocate a fixed amount of storage separately to each user.

**Comment by brl0** [comment]
Another consideration might be to allow for setting quotas for users and groups.
If you shell into the nfs pod (at least on gcp, aws efs might need a separate pod) the setquota command is available.  I think integrating user and group quotas into the qhub config would be a great feature to add.

**Reply by alimanfoo** [reply]
Hi @brl0, I don't suppose you would have any more advice about how to manually set storage quotas? I figured out how to connect to the nfs-server container via `kubectl exec`, and as you said the `setquota` command is available, but I don't think it works because the storage isn't mounted with quotas enabled. I'm way out of my depth here and maybe doing something that isn't advisable, but just wondering if there was a way.

**Comment by dharhas** [comment]
@danlester @costrouc the new way we are considering dealing with shared folders should allow this right?

**Comment by costrouc** [comment]
I agree this is the direction we need to go. Also yes @dharhas the new approach will support this. We are going to explicitly mount each shared folder individually (possible from the same pvc). And the user he me directory will be some filesystem that we mount. One big issue with if we mount the user as a block filesystem is that it will not be a shared filesystem and available to the dask workers.

**Comment by Adam-D-Lewis** [comment]
I'm closing old discussions in a repo cleanup effort.  Feel free to re-open if needed.
